setup.py

```diff
@@ -66,14 +68,13 @@ def run(self):
         envlib = PYENV_PREFIX / 'lib' / SYSLIB[platform.system()]
         os.symlink(envlib, libsaw_prefix / envlib.name)
         os.chdir(PARENT)
-        subprocess.check_call(
-          ["git", "submodule", "deinit", "-f", "submodules/jigsaw-python"])
+        subprocess.check_call(["git", "submodule", "deinit", "-f", "submodules/jigsaw-python"])
 
     def _check_gcc_version(self):
         cpp = shutil.which("c++")
-        major, minor, patch = subprocess.check_output(
-            [cpp, "--version"]
-            ).decode('utf-8').split('\n')[0].split()[-1].split('.')
+        line = subprocess.check_output([cpp, "--version"]).decode('utf-8').split('\n')[0]
+        m = re.search('(\d+\.)?(\d+\.)?(\d+)', line)
+        major, minor, patch = line[m.start(): m.end()].split('.')
```

Comment:
@SanderHulst it's a really good idea to switch to regex, but it doesn't cover `cpp (MacPorts gcc12 12.2.0_2+stdlib_flag) 12.2.0` from #47.  How about using `m = re.search('(\d+\.)(\d+\.)(\d+)', line)` instead? We're assuming there are 3 parts when it's split anyway, so why not make all segments mandatory in the version regex?

Please also add the same in the test you created for this (also thanks for that)

Reply:
Also I'm not sure if it fully covers issues in #47, so maybe we keep that ticket open even after merging this.